src/datasets/info.py

```diff
@@ -189,13 +206,23 @@ def unique(values):
         features = None
         supervised_keys = None
 
+        # Find common task templates across all dataset infos
+        all_task_templates = [info.task_templates for info in dataset_infos if info.task_templates is not None]
+        if len(all_task_templates) > 1:
+            task_templates = list(set(all_task_templates[0]).intersection(*all_task_templates[1:]))
+        elif len(all_task_templates):
+            task_templates = list(set(all_task_templates[0]))
+        else:
+            task_templates = None
```

Comment:
I'd love to see this unit tested, to make sure it actually does what we want, ie select the TaskTemplate(s) that are present in every DatasetInfo.

Reply:
great idea! i've added some tests to cover the various scenarios that can arise